docs: replace exercise for metrics getting-started (#2878)
keptn/lifecycle-toolkit: docs/docs/getting-started/metrics.md

Minimal configuration is required
because Keptn hooks directly into Kubernetes primitives.

The Keptn metrics feature

> integrates metrics from all these sources into a single set of metrics.

Reviewer: ```suggestion
integrates metrics from any number of metrics providers into a single set of metrics.
```

     ```

     For a cleaner display, remove some attributes:

     ```shell
     prometheus_http_requests_total{code="200", container="prometheus", \
          endpoint="http-web", handler="/-/healthy", \
          job="kube-prom-stack-kube-prome-prometheus", namespace="default", \
          service="kube-prom-stack-kube-prome-prometheus"}
     ```

     This is your metric and value as it is stored in Prometheus.

## Configure Keptn for your metrics

Now you must tell Keptn about the metrics you are using.j
You do this by defining:

- A
  [KeptnMetricsProvider](../reference/crd-reference/metricsprovider.md)
  resource to define the external observability platform you are using
  as a data source.
  For this exercise, this is the Prometheus server
  you installed and exposed above.
- A
  [KeptnMetric](../reference/crd-reference/metric.md)
  resource to define each metric query you want to pull.

The steps are:

1. Create a
   [KeptnMetricsProvider](../reference/crd-reference/metricsprovider.md/)
   resource for the observability platform you are using as a data source.
   To do this, create a new `.yml` file with content like the following:

>       ```shell

Reviewer 2: ```suggestion
      ```yaml
```

        apiVersion: metrics.keptn.sh/v1
        kind: KeptnMetricsProvider
        metadata:
          name: local-prometheus
          namespace: default
        spec:
          type: prometheus
          targetServer: "http://prometheus-operated.default.svc.cluster.local:9090/"

>       ```

Reviewer: let's move this to a separate file and use include

Author: Yes, this is one of the items listed in https://github.com/keptn/lifecycle-toolkit/issues/3413 to be done in a separate PR.

     provide information about the token and key.

     Keptn uses the `name` you assign to reference this data source.

     The `targetServer` field tells Keptn where to find this data source;
     in this case, it points to the UI on port 9090.

     Apply this file with the following command:

     ```shell
      kubectl apply -f YOUR-KEPTN-METRIC-PROVIDER.yml
      ```

1. Define your
   [KeptnMetric](../reference/crd-reference/metric.md/)
   custom resource for each piece of data you want to pull
   (in this case, the Prometheus query for the metric you selected).

     This is where you use the Prometheus query from before.
     Note: The namespaces of `KeptnMetricsProvider` and `KeptnMetric` must match.

     ```yaml
     apiVersion: metrics.keptn.sh/v1
     kind: KeptnMetric
     metadata:
       name: prometheus-http-requests-total
       namespace: default
     spec:
       provider:
         name: local-prometheus
       query: 'prometheus_http_requests_total{code="200", handler="/-/healthy", job="kube-prom-stack-kube-prome-prometheus", namespace="default", service="kube-prom-stack-kube-prome-prometheus"}'
       fetchIntervalSeconds: 10

>      ```

Reviewer: let's move this to a separate file and use include

Author: See above.
